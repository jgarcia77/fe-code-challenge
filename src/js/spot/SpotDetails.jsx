import React from 'react';
import PropTypes from 'prop-types';
import Button from '../common/Button';
import {formatPrice} from "../utils/number-formatting";
import {withRouter} from 'react-router-dom';

const SpotDetails = ({spot, history}) => {
    const _onBookItClick = () => {
        history.push('/checkout');
    }

    return (
        <div className="spot-details">
            <h1 className="heading-sm title">Spot Details</h1>
            <h2 className="heading-sm sub-title">{spot.title}</h2>
            <p>{spot.description}</p>
            <div className="book-it">
                <Button data-testid="test-book-it" color="primary" onClick={_onBookItClick}>
                    {formatPrice(spot.price)} &#124; Book It!
                </Button>
            </div>
        </div>
    );
}

SpotDetails.propTypes = {
    spot: PropTypes.shape({
        title: PropTypes.string,
        description: PropTypes.string,
        price: PropTypes.number
    })
};

SpotDetails.defaultProps = {
    spot: {}
};

export default withRouter(SpotDetails);