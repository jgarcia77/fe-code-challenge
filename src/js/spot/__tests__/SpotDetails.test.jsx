import '@testing-library/jest-dom'
import React from 'react';
import {render, fireEvent, screen} from '@testing-library/react';
import SpotDetails from '../SpotDetails';
import renderer from 'react-test-renderer';

const props = {
    id: 1,
    title: "Seven Lions - Valet",
    price: 1400,
    description: "Arrive at this location at 97 E Adams St. This is the Seven Lions valet stand operated by VPA. It is located in front of Starbucks on the south side of E Adams St. between S Michigan Ave. and S Wabash Ave. If you are using the SpotHero Mobile App, please refer to the 2nd picture for an annotated map.",
    distance: "0.1 mi",
    image: "https://res.cloudinary.com/spothero/image/upload/v1579704583/front-end/code-challenge/wkabs86wl5zwqaksb6fr.jpg",
    history: {
        push: jest.fn()
    }
}

const component = <SpotDetails.WrappedComponent {...props} />;

describe("Feature: SpotDetails Component", () => {
    afterEach(() => {
        jest.restoreAllMocks();
    });

    describe("Scenario: Test Markup", () => {
        describe("When the component is rendered", () => {
            it("Then it should match the snapshot", () => {
                const json = renderer.create(component);

                expect(json).toMatchSnapshot();
            });
        });
    });

    describe("Scenario: Test Events", () => {
        describe("When the book it button is clicked", () => {
            it("Then it should navigate to the checkout page", () => {
                render(component);

                const btnBookIt = screen.getByTestId("test-book-it");
            
                fireEvent.click(btnBookIt);
            
                expect(props.history.push).toHaveBeenCalledWith("/checkout");
            });
        });
    });
});