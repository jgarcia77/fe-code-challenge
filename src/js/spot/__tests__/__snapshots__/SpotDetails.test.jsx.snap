// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature: SpotDetails Component Scenario: Test Markup When the component is rendered Then it should match the snapshot 1`] = `
<div
  className="spot-details"
>
  <h1
    className="heading-sm title"
  >
    Spot Details
  </h1>
  <h2
    className="heading-sm sub-title"
  />
  <p />
  <div
    className="book-it"
  >
    <button
      className="Button Button-primary"
      data-testid="test-book-it"
      onClick={[Function]}
      type="button"
    >
      $NaN
       | Book It!
    </button>
  </div>
</div>
`;
